model-optimizer/extensions/front/tf/SSliceComplexRollReplacement.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging as log
+
+from extensions.middle.InsertLayoutPropagationTransposes import mark_input_as_in_correct_layout
+from extensions.ops.roll import Roll
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.front.subgraph_matcher import SubgraphMatch
+from mo.graph.graph import Graph, Node, rename_nodes
+
+
+class SSliceComplexRollReplacement(FrontReplacementSubgraph):
+    """
+    Some TF models contain the sub-graph
+               SomeOp
+                 |
+    --------------------------
+    |                        |
+    StridedSlice          StridedSlice
+    |                       |
+    ------------------------
+         Complex
+          |
+          |   shift   axes
+          |       |      |
+         -------------------
+                 Roll
+
+    Here SomeOp is some node with real output and with the shape [N_0, ..., N_{r - 1}, 2], and StridedSlice nodes
+    have output shapes [N_0, ..., N_{r - 1}].
+
+    But MO and Inference Engine do not support for complex tensors. Hence, we need to replace this sub-graph with
+
+         SomeOp   shift   axes
+          |       |      |
+         -------------------
+                 Roll
+
+    And, in such replacement, we should correct axes of Roll, because input data are real now. Namely, if
+    there are negative axes for Roll, we need subtract 1 from such axes indices.
+
+    The transformation SSliceComplexRollReplacement performs such replacement.
+    """
+    enabled = True
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('strided_slice_real', dict(op='StridedSlice')),
+                ('strided_slice_imag', dict(op='StridedSlice')),
+                ('complex', dict(op='Complex')),
+                ('roll', dict(op='Roll')),
+            ],
+            edges=[
+                ('strided_slice_real', 'complex', {'in': 0}),
+                ('strided_slice_imag', 'complex', {'in': 1}),
+                ('complex', 'roll', {'in': 0}),
+            ])
+
+    def replace_sub_graph(self, graph: Graph, match: [dict, SubgraphMatch]):
+        strided_slice_real = match['strided_slice_real']
+        strided_slice_imag = match['strided_slice_imag']
+
+        real_input = strided_slice_real.in_port(0).get_source().node
+        imag_input = strided_slice_imag.in_port(0).get_source().node
+        if real_input.soft_get('name', real_input.id) != imag_input.soft_get('name', imag_input.id):
```

Comment:
nit: compare `real_input.id` and `imag_input.id` instead of names, because `ids` are guaranteed to be unique, while node names not.

Reply:
Done.